homeassistant/components/mqtt/__init__.py

```diff
@@ -187,7 +188,11 @@
     required=True,
 )
 
-CONFIG_SCHEMA_BASE = vol.Schema(
+PLATFORM_CONFIG_SCHEMA_BASE = vol.Schema(
+    {vol.Optional(component.value): cv.ensure_list for component in PLATFORMS}
```

Comment:
With https://github.com/home-assistant/core/pull/71995 we would instead make this the platform schemas.